util/ranger/points.go

```diff
@@ -361,6 +344,28 @@ func HandlePadCharToFullLength(sc *stmtctx.StatementContext, ft *types.FieldType
 	}
 }
 
+// handleUnsignedIntCol handles negative integer for unsigned integer colunm
+func handleUnsignedIntCol(ft *types.FieldType, val types.Datum, op string) (types.Datum, string, error) {
+	isUnsigned := mysql.HasUnsignedFlag(ft.Flag)
+	isIntegerType := mysql.IsIntegerType(ft.Tp)
+	isNegativeInteger := (val.Kind() == types.KindInt64 && val.GetInt64() < 0)
+
+	if !isUnsigned || !isIntegerType || !isNegativeInteger {
+		return val, op, nil
+	}
+
+	// if the operation is GT, GE or NE, the range is [0, +inf]
+	// otherwise the value can not match any (key, value) pair in tikv storage.
+	if op == ast.GT || op == ast.GE || op == ast.NE {
+		op = ast.GE
+		val.SetUint64(0)
+	} else {
+		return val, op, kv.ErrNotExist
```

Comment:
Why do we return kv.ErrNotExist here?
Can we return a bool flag like `isValidRange` to indicate whether this range is valid or invalid?

Reply:
modified